infiniband-diags/ibportstate.c

```diff
@@ -401,8 +401,8 @@ int main(int argc, char **argv)
 	    "\twidth, query, down, arm, active, vls, mtu, lid, smlid, lmc,\n"
 	    "\tmkey, mkeylease, mkeyprot\n";
 	const char *usage_examples[] = {
-		"3 1 disable\t\t\t# by lid",
-		"-G 0x2C9000100D051 1 enable\t# by guid",
```

Comment:
I don't see why the GUID option needs to be removed from the example.  You do need to specify the HCA/port to get onto the correct fabric (subnet prefix) but it should work.  The code should resolve through resolve_portid_str() -> resolve_guid()?

Reply:
The example "-G 0x2C9000100D051 1 enable\t# by guid" is some kind of misleading. ibportstat user may pass a HCA guid to ibportstat.

How about this patch?
```
diff --git a/infiniband-diags/ibportstate.c b/infiniband-diags/ibportstate.c
index 17d1e1503834..6dab1ea09630 100644
--- a/infiniband-diags/ibportstate.c
+++ b/infiniband-diags/ibportstate.c
@@ -401,8 +401,9 @@ int main(int argc, char **argv)
 	    "\twidth, query, down, arm, active, vls, mtu, lid, smlid, lmc,\n"
 	    "\tmkey, mkeylease, mkeyprot\n";
 	const char *usage_examples[] = {
-		"3 1 disable\t\t\t# by lid",
-		"-G 0x2C9000100D051 1 enable\t# by guid",
+		"-C qib0 -P 1 -G 0x2C9000100D051 1 enable\t# by CA name, CA Port Number, switch guid, physical port number",
+		"-C qib0 -P 1 3 1 disable     # by CA name, CA Port Number, lid, physical port number",
+		"-C qib0 -P 1 3 1 enable      # by CA name, CA Port Number, lid, physical port number",
 		"-D 0 1\t\t\t# (query) by direct route",
 		"3 1 reset\t\t\t# by lid",
 		"3 1 speed 1\t\t\t# by lid",
diff --git a/infiniband-diags/man/ibportstate.8.in.rst b/infiniband-diags/man/ibportstate.8.in.rst
index da1de4c76317..3d5604b280e6 100644
--- a/infiniband-diags/man/ibportstate.8.in.rst
+++ b/infiniband-diags/man/ibportstate.8.in.rst
@@ -22,8 +22,9 @@ DESCRIPTION
 ibportstate allows the port state and port physical state of an IB port
 to be queried (in addition to link width and speed being validated
 relative to the peer port when the port queried is a switch port),
-or a switch port to be disabled, enabled, or reset. It
-also allows the link speed/width enabled on any IB port to be adjusted.
+or a switch port to be disabled, enabled, or reset. InfiniBand HCA port
+state may be changed locally without the knowledge of the Subnet Manager.
+It also allows the link speed/width enabled on any IB port to be adjusted.
 
 OPTIONS
 =======
@@ -34,8 +35,8 @@ OPTIONS
                         mkey, mkeylease, mkeyprot
 			(Default is query)
 
-        **enable, disable, and reset** are only allowed on switch ports (An
-        error is indicated if attempted on CA or router ports)
+        **enable, disable, and reset** change or reset a switch or HCA port state
+        (You must specify the CA name and Port number when locally change CA port state.)
 
         **off** change the port state to disable.
 
@@ -114,8 +115,9 @@ EXAMPLES
 ========
 
 ::
-        ibportstate 3 1 disable                  # by lid
-        ibportstate -G 0x2C9000100D051 1 enable  # by guid
+        ibportstate -C qib0 -P 1 -G 0x2C9000100D051 1 enable # by CA name, CA Port Number, switch guid, physical port number
+        ibportstate -C qib0 -P 1 3 1 disable     # by CA name, CA Port Number, lid, physical port number
+        ibportstate -C qib0 -P 1 3 1 enable      # by CA name, CA Port Number, lid, physical port number
         ibportstate -D 0 1                       # (query) by direct route
         ibportstate 3 1 reset                    # by lid
         ibportstate 3 1 speed 1                  # by lid
```

Reply:
I don't see any harm in merging proposed patch, if anyone wants "to fix" ibportstat, he is wellcomed to send patches.
The patch from Honggang doesn't break anything.